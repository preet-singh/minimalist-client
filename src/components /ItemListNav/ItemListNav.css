
.ItemListNav_list {
    list-style-type: none;
    margin: 130px 0 0;
    padding: 0;
  }
  
  .ItemListNav_button-wrapper {
    margin: 0 0 0 20px;
    max-width: 170px;
  }
  
  .ItemListNav_add-inventory-button {
    margin: 20px auto;
  }
  .ItemListNav_del-inventory-button {
    margin: 20px auto;
  }
  .ItemListNav_add-inventory-button:hover {
    background-color: #FFB72F;
    color: white;
  }
  .ItemListNav_del-inventory-button:hover {
    background-color: #ff0000;
    color: white;
  }
  
  .ItemListNav_inventory-link {
    background-color: #00264d;
    color: #b5b3b3;
    display: block;
    font-family: var(--custom-font);
    font-size: 20px;
    margin: 0 0 8px auto;
    max-width: 180px;
    padding: 8px 16px;
    text-decoration: none;
  }
  .ItemListNav_inventory-link:hover {
    color: #FFB72F;
    cursor: pointer;
  }
  .ItemListNav_inventory-link.active {
    background-color: #102541;
  }
  
  .ItemListNav_num-items {
    color: #adadad;
    display: block;
    float: right;
    font-size: 10px;
  }
  