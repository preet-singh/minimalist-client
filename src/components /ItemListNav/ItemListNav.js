import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Button from '../Button/Button';
import InventoryContext from '../../InventoryContext';
import { countItemsForInventory } from '../item-helpers';
import config from '../../config';
import PropTypes from 'prop-types';
import './ItemListNav.css';

export default class ItemListNav extends React.Component {
  static defaultProps = {
    onDeleteInventory: () => {},
    match: {
      params: {}
    }
  }
  static contextType = InventoryContext;

  handleDeleteInventory = (e) => {
    e.preventDefault();
    const inventoryId = Number(this.props.match.params.inventoryId)
    console.log(inventoryId);

    fetch(`${config.API_ENDPOINT}/inventory/${inventoryId}`, {
      method: "DELETE",
      headers: {
        "content-type": "application/json"
      },
    })
      .then(res => {
        if (!res.ok) return res.json().then(e => Promise.reject(e));
        else return undefined;
      })
      .then(() => {
        this.context.deleteInventory(inventoryId)
        this.props.onDeleteInventory(inventoryId)
        this.props.history.push('/');
      })
      .catch(error => {
        console.error({ error });
      });
  };
  render() {
    const { inventory=[], items=[] } = this.context
    console.log(inventory);
    return (
      <div className='ItemListNav'>
        <ul className='ItemListNav_list'>
          {inventory.map(inventory =>
            <li key={Number(inventory.id)}>
              <NavLink
                className='ItemListNav_inventory-link'
                to={`/inventory/${inventory.id}`}
              >
                <span className='ItemListNav_num-items'>
                  {countItemsForInventory(items, inventory.id)}
                </span>
                {inventory.inventory_name}
              </NavLink>
            </li>
          )}
        </ul>
        <div className='ItemListNav_button-wrapper'>
          <Button
            tag={Link}
            to='/add-inventory'
            type='button'
            className='ItemListNav_add-inventory-button'
          >
            <FontAwesomeIcon icon='plus' /> 
            <br />
            Inventory
          </Button>

          <Button
            tag={Link}
            to={`/inventory`}
            type='button'
            className='ItemListNav_del-inventory-button'
            onClick={this.handleDeleteInventory}
          >
            <FontAwesomeIcon icon='minus' /> 
            <br />
            Inventory
          </Button>
        </div>
        {/* <div className='ItemListNav_button-wrapper-del' key={Number(inventory.id)} >
          <Button
            tag={Link}
            to={`/inventory`}
            type='button'
            className='ItemListNav_del-inventory-button'
            onClick={this.handleDeleteInventory}
          >
            <FontAwesomeIcon icon='minus' /> 
            <br />
            Inventory
          </Button>
        </div> */}
      </div>
    )
  }
}

ItemListNav.propTypes = {
  onDeleteInventory: PropTypes.func.isRequired,
  match: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired
}
