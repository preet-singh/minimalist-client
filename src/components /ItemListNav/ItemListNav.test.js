import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ItemListNav from './ItemListNav'

describe(`NoteListNav component`, () => {
  it('renders a .ItemListNav by default', () => {
    const wrapper = shallow(<ItemListNav />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  // enzyme doesn't yet support React.createContext
  it('renders a link in ul for each inventory in array', () => {
    const context = {
      items: [
        {
          "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Dogs",
          "inventoryId": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
          "description": "Corporis accusamus placeat.\n \rUnde."
        },
        {
          "id": "d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Cats",
          "inventoryId": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
          "description": "Eos\n \rlaudantium."
        },
        {
          "id": "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Pigs",
          "inventoryId": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
          "description": "Occaecati dignissimos\nvoluptatum nihil."
        },
        {
          "id": "d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Birds",
          "inventoryId": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
          "description": "Eum culpa odit."
        },
      ],
      inventory: [
        {
          "id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Clothes"
        },
        {
          "id": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Bedding"
        },
        {
          "id": "b07162f0-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Kitchen"
        }
      ]
    }
    const ul = shallow(<ItemListNav />, context)
      .find('ul')
    expect(toJson(ul)).toMatchSnapshot()
  })
})
