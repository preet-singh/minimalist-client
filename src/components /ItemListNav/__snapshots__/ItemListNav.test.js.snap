// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteListNav component renders a .ItemListNav by default 1`] = `
<div
  className="ItemListNav"
>
  <ul
    className="ItemListNav_list"
  />
  <div
    className="ItemListNav_button-wrapper"
  >
    <NavButton
      className="ItemListNav_add-inventory-button"
      tag={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      to="/add-inventory"
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="plus"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <br />
      Inventory
    </NavButton>
  </div>
  <div
    className="ItemListNav_button-wrapper"
    key="NaN"
  >
    <NavButton
      className="ItemListNav_del-inventory-button"
      onClick={[Function]}
      tag={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      to="/inventory"
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="minus"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <br />
      Inventory
    </NavButton>
  </div>
</div>
`;

exports[`NoteListNav component renders a link in ul for each inventory in array 1`] = `
<ul
  className="ItemListNav_list"
/>
`;
